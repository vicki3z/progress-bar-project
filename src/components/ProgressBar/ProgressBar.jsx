import React from "react";

const ProgressBar = ({ progressValue }) => (
  <div>
    <span>{progressValue}</span>
  </div>
);

export default ProgressBar;
