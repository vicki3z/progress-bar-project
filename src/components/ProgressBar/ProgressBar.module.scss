.progressBar {
  background-color: #fafafa;
  border-radius: 40px;
  margin-top: 10px;
  position: relative;

  .progress {
    background-color: $secondary-color;
    height: 30px;
    border-radius: 40px;
    vertical-align: middle;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: 0.5s ease-out;

    &.overLimit {
      background-color: #ff5252;
    }
  }

  &Value {
    position: absolute;
    top: 25%;
  }
}

.hide {
  display: none;
}
