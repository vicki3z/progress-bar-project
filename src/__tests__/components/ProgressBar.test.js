import React from "react";
import { shallow } from "enzyme";

import ProgressBar from "../../components/ProgressBar";

describe("Progress Bar Component", () => {
  it("should render without error", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it("should show progress value that has been passed through progressValue property", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.find("span[data-progress='progress-value']")).toHaveLength(
      1
    );
    expect(
      Number(component.find("span[data-progress='progress-value']").text())
    ).toEqual(20);
  });

  it("should show progress value that has been passed through progressValue property", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.find("span[data-progress='progress-value']")).toHaveLength(
      1
    );
    expect(
      Number(component.find("span[data-progress='progress-value']").text())
    ).not.toEqual(10);
  });

  it("should show limit value that has been passed through limit property", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.find("span[data-progress='limit-value']")).toHaveLength(1);
    expect(
      Number(component.find("span[data-progress='limit-value']").text())
    ).toEqual(200);
  });

  it("should have calculatePercentage function", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.instance().calculatePercentage()).toBeDefined();
  });

  it("calculatePercentage of progress 10 and limit 200 should equal to 5", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.instance().calculatePercentage(10, 200)).toEqual(5);
  });

  it("calculatePercentage of progress 62 and limit 230 should equal to 27", () => {
    const component = shallow(<ProgressBar progressValue={20} limit={200} />);
    expect(component.instance().calculatePercentage(62, 230)).toEqual(27);
  });
});
