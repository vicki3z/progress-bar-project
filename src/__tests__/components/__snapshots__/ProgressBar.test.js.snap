// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar Component should render without error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    limit={200}
    progressValue={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "width": "10%",
            }
          }
        />,
        <span
          data-progress="progress-value"
        >
          20
        </span>,
        <span
          data-progress="limit-value"
        >
          200
        </span>,
        <span
          data-progress="percentage"
        >
          10
          %
        </span>,
      ],
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "style": Object {
            "width": "10%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 20,
          "className": undefined,
          "data-progress": "progress-value",
        },
        "ref": null,
        "rendered": 20,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 200,
          "className": undefined,
          "data-progress": "limit-value",
        },
        "ref": null,
        "rendered": 200,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            10,
            "%",
          ],
          "className": undefined,
          "data-progress": "percentage",
        },
        "ref": null,
        "rendered": Array [
          10,
          "%",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "width": "10%",
              }
            }
          />,
          <span
            data-progress="progress-value"
          >
            20
          </span>,
          <span
            data-progress="limit-value"
          >
            200
          </span>,
          <span
            data-progress="percentage"
          >
            10
            %
          </span>,
        ],
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "style": Object {
              "width": "10%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 20,
            "className": undefined,
            "data-progress": "progress-value",
          },
          "ref": null,
          "rendered": 20,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 200,
            "className": undefined,
            "data-progress": "limit-value",
          },
          "ref": null,
          "rendered": 200,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              10,
              "%",
            ],
            "className": undefined,
            "data-progress": "percentage",
          },
          "ref": null,
          "rendered": Array [
            10,
            "%",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
